frame/support/procedural/src/lib.rs

```diff
@@ -583,3 +584,646 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
+/// outside of a pallet module
+fn pallet_macro_stub() -> TokenStream {
+	quote!(compile_error!(
+		"This attribute can only be used from within a pallet module marked with `#[frame_support::pallet]`"
+	))
+	.into()
+}
+
+/// The mandatory attribute `#[pallet::config]` defines the configurable options for the pallet.
+///
+/// Item must be defined as:
+///
+/// ```ignore
+/// #[pallet::config]
+/// pub trait Config: frame_system::Config + $optionally_some_other_supertraits
+/// $optional_where_clause
+/// {
+/// ...
+/// }
+/// ```
+///
+/// I.e. a regular trait definition named `Config`, with the supertrait
+/// `frame_system::pallet::Config`, and optionally other supertraits and a where clause.
+///
+/// The associated type `RuntimeEvent` is reserved. If defined, it must have the bounds
+/// `From<Event>` and `IsType<<Self as frame_system::Config>::RuntimeEvent>`.
+///
+/// See [`pallet::event`](`macro@event`) for more information.
+#[proc_macro_attribute]
+pub fn config(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// The `#[pallet::constant]` attribute can be used to add an associated type trait bounded by `Get` from
+/// [`pallet::config`](`macro@config`) into metadata, e.g.:
+///
+/// ```ignore
+/// #[pallet::config]
+/// pub trait Config: frame_system::Config {
+/// 	#[pallet::constant]
+/// 	type Foo: Get<u32>;
+/// }
+/// ```
+#[proc_macro_attribute]
+pub fn constant(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// To bypass the `frame_system::Config` supertrait check, use the attribute
+/// `pallet::disable_frame_system_supertrait_check`, e.g.:
+///
+/// ```ignore
+/// #[pallet::config]
+/// #[pallet::disable_frame_system_supertrait_check]
+/// pub trait Config: pallet_timestamp::Config {}
+/// ```
+#[proc_macro_attribute]
+pub fn disable_frame_system_supertrait_check(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// To generate a `Store` trait associating all storages, use the attribute
+/// `#[pallet::generate_store($vis trait Store)]`, e.g.:
+///
+/// ```ignore
+/// #[pallet::pallet]
+/// #[pallet::generate_store(pub(super) trait Store)]
+/// pub struct Pallet<T>(_);
+/// ```
+/// More precisely, the `Store` trait contains an associated type for each storage. It is
+/// implemented for `Pallet` allowing access to the storage from pallet struct.
+///
+/// Thus when defining a storage named `Foo`, it can later be accessed from `Pallet` using
+/// `<Pallet as Store>::Foo`.
+#[proc_macro_attribute]
+pub fn generate_store(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// To generate the full storage info (used for PoV calculation) use the attribute
+/// `#[pallet::generate_storage_info]`, e.g.:
+///
+/// ```ignore
+/// #[pallet::pallet]
+/// #[pallet::generate_storage_info]
+/// pub struct Pallet<T>(_);
+/// ```
+///
+/// This requires all storage items to implement the trait `StorageInfoTrait`, thus all keys
```

Comment:
Maybe link to the actual trait:

```suggestion
/// This requires all storage items to implement the trait [`frame_support::traits::StorageInfoTrait`], thus all keys
```

Reply:
can't for this one, see my other comment. We're in `frame_support_procedural` so no traits from frame_support are available